native-code/ios/index.md

```diff
@@ -45,69 +52,54 @@ to update the code in your checkout.
-
-The different output directories can be replaced with any directory of your own
-choice. When running GN; make sure your current working directory is src/ of
-your workspace, then run:
-
-~~~~~ bash
-gn gen out/Debug-device-arm32 --args='target_os="ios" target_cpu="arm" is_component_build=false'
-~~~~~
-
-#### Targeting 64-bit iOS devices
-
-~~~~~ bash
-gn gen out/Debug-device-arm64 --args='target_os="ios" target_cpu="arm64" is_component_build=false'
-~~~~~
-
-**NOTICE:** To build for devices, you must have a signing identity configured
-for your Xcode installation. You can check this by running:
+The variables you should care about are the following:
+
+* `target_os`:
+  - To build for iOS this should be set as `target_os="ios"` in your `gn args`.
+  The default is whatever OS you are building on.
+* `target_cpu`:
+  - For builds targeting iOS devices, this should be set to either `"arm"` or
+  `"arm64"`, depending on the architecture of the device. For builds to run in
+  the simulator, this should be set to `"x64"`.
+* `is_component_build`:
+  - Component builds don't take as long to link, but have runtime performance
+  implications. They are not supported on iOS, so this should always be set
+  to `false`.
+* `is_debug`:
+  - Debug builds are the default. When building for release, specify `false`.
 
-~~~~~ bash
-python build/config/ios/find_signing_identity.py
-~~~~~
-
-If you need to build for ARM devices without this setup, you can add the
-`ios_enable_code_signing=false` variable to GN's arguments.
-
-#### Targeting the iOS Simulator
-
-~~~~~ bash
-gn gen out/Debug-sim32 --args='target_os="ios" target_cpu="x86" is_component_build=false'
-~~~~~
+The component build is the default for Debug builds, which are also enabled by
+default unless `is_debug=false` is specified.
 
-#### Targeting 64-bit iOS Simulator
+The [GN][5] command for generating build files is `gn gen <output folder>`.
 
+#### Examples
 ~~~~~ bash
-gn gen out/Debug-sim64 --args='target_os="ios" target_cpu="x64" is_component_build=false'
-~~~~~
+# debug build for 64-bit iOS
+gn gen out/ios_64 --args='target_os="ios" target_cpu="arm64" is_component_build=false'
 
-#### Targeting macOS
-
-The host OS is the default `target_os`, so you don't need to specify it:
-
-~~~~~ bash
-gn gen out/Debug-mac
+# debug build for simulator
+gn gen out/ios_sim --args='target_os="ios" target_cpu="x64" is_component_build=false'
 ~~~~~
 
 ### Compiling with ninja
 
 To compile, just run ninja on the appropriate target. For example:
 
 ~~~~~ bash
-ninja -C out/Debug-device-arm32 AppRTCMobile
+ninja -C out/ios_64 AppRTCMobile
 ~~~~~
 
-Some sample scripts are also available in [webrtc/build/ios][3].
+Replace `AppRTCMobile` in the command above with the target you
+are interested in.
 
+To see a list of available targets, run `gn ls out/<output folder>`.
 
-### Compiling with Xcode
+### Using Xcode
+
+GN can generate an Xcode project from the build files. The build targets
```

Comment:
I think this whole section can be bit more structured.
I propose two parts:
1. Generating and accessing the workspace. 
i.e 
*GN can generate Xcode workspace when generating the build files. To do so, use the --ide=xcode argument in the GN command. When running the GN script with this argument, Xcode workspace "all.xcworkspace" will be created in the output directory. Note that the other GN arguments you provide will have effect on the Xcode project as well.
For instance the following GN command* 
`gn gen out/ios_64 --ide=xcode --args='target_os="ios" target_cpu="x64" is_component_build=false`
*will create all.xcworkspace in the out/ios_64 directory. The target os will be iOS and the target cpu will be arm64.*